### PR TITLE
Ability to @derive StathamLogger.Loggable

### PR DESCRIPTION
**Pull request by YuriiZdyrko**

### CODE REVIEW

File: lib/loggable.ex [hunk 1]
```diff
@@ -36,6 +53,10 @@ defimpl StathamLogger.Loggable, for: Atom do
   def sanitize(data, _), do: data
 end
 
+defimpl StathamLogger.Loggable, for: PID do
```
Comment (fuelen) [inline lib/loggable.ex:56]: Let's define for `Reference` too

File: test/statham_logger_test.exs [hunk 2]
```diff
@@ -14,12 +14,48 @@ defmodule StathamLoggerTest do
         StathamLogger,
         device: :user,
         level: nil,
-        metadata: []
+        metadata: [
+          sensitive_keys: [:password],
```
Comment (fuelen) [inline test/statham_logger_test.exs:18]: I suggest an API that is used in Phoenix for filtering params:
rename `sensitive_keys` to `filter_keys` and use `{:keep | :discard, [atom()}` api
```elixir
metadata: [
  filter_keys: {:discard, [:password]}
  # ...
]
```

File: test/support/loggable_struct.ex [hunk 3]
```diff
@@ -0,0 +1,5 @@
+defmodule StathamLogger.LoggableStruct do
+  @derive {StathamLogger.Loggable, sensitive_keys: [:phone_number]}
```
Comment (fuelen) [inline test/support/loggable_struct.ex:2]:
```elixir
  @derive {StathamLogger.Loggable, discard: [:phone_number]}
```

**Review by fuelen — CHANGES_REQUESTED** [review]

File: test/statham_logger_test.exs (hunk 2 above)
Comment (YuriiZdyrko) [reply test/statham_logger_test.exs]: Done. I was thinking about something similar.

File: lib/loggable.ex [hunk 4]
```diff
@@ -62,9 +97,15 @@ defimpl StathamLogger.Loggable, for: Map do
   defp sanitize_map_key(key) when is_binary(key) or is_atom(key) or is_number(key), do: key
   defp sanitize_map_key(key), do: inspect(key)
 
-  defp should_be_filtered(field, sensitive_keys) do
-    Enum.any?(sensitive_keys, fn key -> to_string(key) == to_string(field) end)
+  defp should_be_filtered(field, {:keep, keys = [_h | _t]}) do
+    Enum.any?(keys, fn key -> to_string(key) != to_string(field) end)
```
Comment (fuelen) [inline lib/loggable.ex:101]: let's stringify fileld outside the loop

File: lib/loggable.ex [hunk 5]
```diff
@@ -62,9 +97,15 @@ defimpl StathamLogger.Loggable, for: Map do
...
   end
+
+  defp should_be_filtered(field, {:discard, keys = [_h | _t]}) do
+    Enum.any?(keys, fn key -> to_string(key) == to_string(field) end)
+  end
```
Comment (fuelen) [inline lib/loggable.ex:106]: Alternative implementation for this function:
```elixir
  defp should_be_filtered(field, {operation, keys}) when is_list(keys) do
  comparator =
    case operation do
      :keep -> &==/2
      :discard -> &!=/2
    end

  field = to_string(field)
  Enum.any?(keys, fn key -> comparator.(to_string(key), field) end)
end
```

File: lib/loggable.ex [hunk 6]
```diff
@@ -33,25 +54,39 @@ defimpl StathamLogger.Loggable, for: Any do
...
 defimpl StathamLogger.Loggable, for: Integer do
+  @impl true
   def sanitize(data, _), do: data
 end
 
 defimpl StathamLogger.Loggable, for: Map do
+  @impl true
   def sanitize(map, opts) do
-    sensitive_keys = Keyword.get(opts, :sensitive_keys) || []
+    filter_keys = Keyword.get(opts, :filter_keys)
```
Comment (fuelen) [inline lib/loggable.ex:84]: Stringify keys in this place

File: lib/statham_logger.ex [hunk 7]
```diff
@@ -47,25 +47,32 @@ defmodule StathamLogger do
       a confirmation is received.
 
   ## Implementing sanitization for new types can be done in 2 ways:
```
Comment (fuelen) [inline lib/statham_logger.ex:49]: in two ways, but there are 3 described. Let's note, that 3rd type is not preferred

File: lib/loggable.ex (hunk 5 above)
Comment (YuriiZdyrko) [reply lib/loggable.ex]: My code had a bug - I had to use `Enum.all?` for `keep` and update test.

**Review by fuelen — APPROVED** [review]

**Pull request merged**